File: app/src/main/java/com/wya/env/base/BaseMvpActivity.java
package com.wya.env.base;

import android.app.Activity;
import android.content.Intent;

import com.wya.env.MainActivity;

/**
 * @date: 2018/7/3 13:48
 * @author: Chunjiang Mao
 * @classname: BaseMvpActivity
 * @describe: BaseMvpActivity
 */

public abstract class BaseMvpActivity<T extends BasePresent> extends BaseActivity implements BaseView {
    
    /**
     * 显示加载对话框
     */
    @Override
    public void showLoading() {
        loadingDialog.show();
    }
    
    /**
     * 隐藏加载对话框
     */
    @Override
    public void hideLoading() {
        loadingDialog.dismiss();
    }
    
    /**
     * 失败回调
     *
     * @param s
     */
    @Override
    public void failedResult(String s) {
        showShort(s);
    }
    
    /**
     * token失效
     */
    @Override
    public void tokenFaile(Activity activity) {
        //跳转到登陆界面
        startActivity(new Intent(activity, MainActivity.class));
        activity.finish();
    }
}

File: app/src/main/java/com/wya/env/base/BaseMvpActivity.kt
package com.wya.env.base

import android.app.Activity
import android.content.Intent

import com.wya.env.MainActivity

/**
 * @date: 2018/7/3 13:48
 * @author: Chunjiang Mao
 * @classname: BaseMvpActivity
 * @describe: BaseMvpActivity
 */

abstract class BaseMvpActivity<T : BasePresent<*>> : BaseActivity(), BaseView {

    /**
     * 显示加载对话框
     */
    override fun showLoading() {
        loadingDialog.show()
    }

    /**
     * 隐藏加载对话框
     */
    override fun hideLoading() {
        loadingDialog.dismiss()
    }

    /**
     * 失败回调
     *
     * @param s
     */
    override fun failedResult(s: String) {
        showShort(s)
    }

    /**
     * token失效
     */
    override fun tokenFaile(activity: Activity) {
        //跳转到登陆界面
        startActivity(Intent(activity, MainActivity::class.java))
        activity.finish()
    }
}
